core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -183,4 +183,13 @@ public void onReady() {}
   public boolean isReady() {
     return true;
   }
+
+  /**
+   * Enables per-message compression, if an encoding type has been negotiated.  If no message
+   * encoding has been negotiated, this is a no-op.
```

Comment:
The default value is de-jure Undefined but de-facto off.  It would probably be better if it was on by default for android (since bandwidth is a concern) and off for datacenter clients.  It could even be selectively enabled based on recent previous call size and speed, so I am intentionally leaving it vague for now.  

If the encoding type is negotiated after for before it shouldn't matter.  This call affects what sentMessage will do, which is strictly after negotiation.  Calling this before start will probably break since stream will not yet exist.